src/NuGet.Core/NuGet.Commands/RestoreCommand/ContentFiles/ContentFileUtils.cs

```diff
@@ -61,6 +62,19 @@ internal static class ContentFileUtils
                 if (nearestGroup != null)
                 {
                     groups.Add(nearestGroup);
+
+                    if (maccatalystFallback != null)
```

Comment:
nit: There's 3 copies of this method. Using a shared version might reduce the possibility of typos.